OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -102,12 +112,24 @@ void INotifyCreated.Created(Actor self)
 
 		void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barrel)
 		{
-			if (a != null && a.Info.AmmoPoolName == Info.Name)
+			if (a != null && Info.Armaments.Contains(a.Info.Name))
 				TakeAmmo();
 		}
 
 		void INotifyAttack.PreparingAttack(Actor self, Target target, Armament a, Barrel barrel) { }
 
+		void ITick.Tick(Actor self)
```

Comment:
You can save running this every tick by renaming it to something like `UpdateEmptyCondition` and calling it only from `GiveAmmo` and `TakeAmmo`.